src/mesh/distributed_mesh.C

```diff
@@ -60,14 +60,30 @@ DistributedMesh::DistributedMesh (const Parallel::Communicator & comm_in,
 
 DistributedMesh & DistributedMesh::operator= (DistributedMesh && other_mesh)
 {
-  // First set this mesh equal to other_mesh as an UnstructuredMesh.
+  // First make a call to MeshBase::clear
+  this->MeshBase::clear();
+
+  // Move assign as a ParallelObject.
+  this->ParallelObject::operator=(other_mesh);
+
+  // Move assign as a MeshBase.
+  this->MeshBase::operator=(std::move(other_mesh));
```

Comment:
In theory, shouldn't this also do the ParallelObject::operator= ?  In the normative sense of "should", at least.